The-Academic-Observatory/oaebu-workflows
# astro migration

# Astro Migration

This PR refactors the codebase for use with The Astronomer Airflow service. Broadly speaking, the changes are as follows:
- Change DAG creation to use the Airflow Taskflow API
- Upgrade to Ariflow 2.7.3
- Removed setup.py and setup.cfg - oaebu-workflows is no longer an installable package
- Created Dockerfile, which inherits from the Astro airflow image. Used to ship to Astro deployment
- Move all workflow files (and their dependent files) into the dag folder (ready for Astro shipping)
- Move tests to the ./tests folder
- Update the API usage to the new Dataset API (https://github.com/The-Academic-Observatory/observatory-platform/pull/653) that uses BigQuery as a storage service for Release objects
- Update the workflow Release objects to be passed as Xcoms between tasks (a benefit of the taskflow upgrade)
- Collapse the upload tasks into their respective file creation tasks. i.e. upload_downloaded() is absorbed into download(). This is necessary now as we no longer necessarily have persistent disks in Astro deployments.
- Bigquery tasks create the client on behalf of their appropriate publisher, rather than have them all invoke the default environment project.
- Updated telescopes that can have multiple releases to use the new dynamic tasks functionality of the taskflow API
